File: TakeThePill/app/src/main/java/embeddedproject/takethepill/CustomAdapterDrug.java
package embeddedproject.takethepill;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;


import java.util.ArrayList;

public class CustomAdapterDrug extends ArrayAdapter<DrugEntity> {

    private ArrayList<DrugEntity> dataSet;
    Context mContext;

    public CustomAdapterDrug(ArrayList<DrugEntity> data, Context context) {
        super(context, R.layout.row_item_drug, data);
        this.dataSet = data;
        this.mContext=context;

    }

    @SuppressLint("InflateParams")
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) getContext()
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        convertView = inflater.inflate(R.layout.row_item_drug, null);

        TextView name = convertView.findViewById(R.id.drugName);
        TextView type = convertView.findViewById(R.id.tvType);
        TextView scorte = convertView.findViewById(R.id.tvScorte);
        TextView prezzo = convertView.findViewById(R.id.tvPrezzo);
        TextView descr = convertView.findViewById(R.id.drugDescr);

        DrugEntity actual = getItem(position);

        name.setText(actual.getNome());
        type.setText("Tipo: "+actual.getTipo());
        scorte.setText("Scorte: " + actual.getScorte());
        prezzo.setText("Prezzo: " +actual.getPrezzo()+ "€");
        if(actual.getDescrizione().equals("")||actual.getDescrizione()==null)
            descr.setText("Nessuna descrizione");
        else
        descr.setText("Descrizione: " + actual.getDescrizione());

        return convertView;
    }


}

File: TakeThePillKV/app/src/main/java/embeddedproject/takethepillkv/CustomAdapterDrug.kt
package embeddedproject.takethepillkv

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import java.util.ArrayList

class CustomAdapterDrug(private val dataSet: ArrayList<DrugEntity>, internal var mContext: Context?) : ArrayAdapter<DrugEntity>(mContext, R.layout.row_item_drug, dataSet) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val inflater = context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        convertView = inflater.inflate(R.layout.row_item_drug, null)

        //Definizione costanti
        val name = convertView!!.findViewById<View>(R.id.drugName) as TextView
        val type = convertView.findViewById<View>(R.id.tvType) as TextView
        val scorte = convertView.findViewById<View>(R.id.tvScorte) as TextView
        val prezzo = convertView.findViewById<View>(R.id.tvPrezzo) as TextView
        val descr = convertView.findViewById<View>(R.id.drugDescr) as TextView

        val actual = getItem(position)

        //Memorizzazione nelle costanti dei valori ritornati dai metodi get
        name.text = actual!!.getNome()
        type.text = "Tipo: " + actual.getTipo()
        scorte.text = "Scorte: " + actual.getScorte()
        prezzo.text = "Prezzo: " + actual.getPrezzo() + "€"

        //Gestione campo descrizione
        if (actual.getDescrizione().equals("") || actual.getDescrizione() == null)
            descr.text = "Nessuna descrizione"
        else
            descr.text = "Descrizione: " + actual.getDescrizione()

        return convertView
    }


}
